### ISSUE TITLE
Allow custom management of equals() and hashCode()

### ISSUE THREAD

**Issue by marcosalis** [LachlanMcKee/gsonpath#49]
This is merely an enhancement proposal, but it would be nice to have.
Sometimes, you would need to de/serialize (and cache) data models coming from a server, which are uniquely identified by their ID. In this case, you want to make sure equals() and hashCode() are consistent with this business logic, so that models having the same ID are equals in the client code as well.
As far as I've seen this is not possible with Gson Path, and it's not very well supported even in Google AutoValue. Is it theoretically possible to allow this, perhaps subclassing an abstract class with the @AutoGsonAdapter annotation, where equals() and hashCode() (and potentially other methods or transient fields) are implemented?

**Comment by LachlanMcKee** [comment]
Hi @marcosalis,

Thanks for opening a ticket. I'm a little unsure what you are asking, so do you mind clarifying?

There should be nothing stopping you writing your own `equals()` and `hashCode()` method inside of a class annotated with `@AutoGsonAdapter`. Did you want the library to generate this for you? If so you could use the annotate an interface. There is a guide for this in the root of the project.

At the moment, the interface will generate `equals()` and `hashCode()` methods that check every single property, as opposed to you being able to exclude certain properties. It might be very useful to be able to exclude certain fields.

**Comment by marcosalis** [comment]
Hi @LachlanMcKee, thank you for your reply! I assumed that the `equals()/hashCode()` methods would have been generated even when annotating a class which already has an implementation. If that's not the case, the library already behaves as desired for me.
It would be useful indeed to annotate methods of an interface so that a represented fields won't be included in the equality computation.

**Comment by LachlanMcKee** [comment]
Hi @marcosalis.

Your first point is correct, the `equals()` and `hashCode()` implementations won't be overridden. Is there a part of the documentation that gave you this impression? I may need to reword it to be clearer.

I will certainly investigate adding support for manipulating the interfaces generated `equals()` and `hashCode()` methods!

**Comment by LachlanMcKee** [comment]
Closing this issue as it is no longer on the road map at this stage.
